src/FakeItEasy/Configuration/AfterMustHaveHappenedConfiguration.cs

```diff
@@ -17,11 +17,38 @@ public AfterMustHaveHappenedConfiguration(FakeManager fakeManager, ICallMatcher
             this.repeatConstraint = repeatConstraint;
         }
 
-        public void InOrder(ISequentialCallContext context)
+        /// <summary>
+        /// Checks that the asserted call happened in order relative to others in the assertion chain.
+        /// </summary>
+        /// <param name="nextAssertion">An assertion describing the next call that should occur.</param>
+        /// <returns>An object that can be used to assert that a following call was made in the expected order.</returns>
+        /// <exception cref="ExpectationException">The call was not made in the expected order.</exception>
+        public IOrderableCallAssertion Then(IAfterMustHaveHappenedConfiguration nextAssertion)
         {
-            Guard.AgainstNull(context, "context");
+            var context = ServiceLocator.Current.Resolve<SequentialCallContext>();
+            this.CheckCallHappenedInOrder(context);
+            return new SubsequentOrderedCallAssertion(context).Then(nextAssertion);
+        }
 
+        private void CheckCallHappenedInOrder(SequentialCallContext context)
+        {
             context.CheckNextCall(this.fakeManager, this.matcher.Matches, this.callDescription, this.repeatConstraint);
         }
+
+        private class SubsequentOrderedCallAssertion : IOrderableCallAssertion
+        {
+            private readonly SequentialCallContext context;
+
+            public SubsequentOrderedCallAssertion(SequentialCallContext context)
+            {
+                this.context = context;
+            }
+
+            public IOrderableCallAssertion Then(IAfterMustHaveHappenedConfiguration nextCall)
+            {
+                ((AfterMustHaveHappenedConfiguration)nextCall).CheckCallHappenedInOrder(this.context);

```

Comment:
> Something's off here. Between the public invocation point (IAfterMustHaveHappenedConfiguration.Then()) and this cast, there's nothing which guarantees that this cast will work. I guess CheckCallHappenedInOrder isn't appropriate to be public on IAfterMustHaveHappenedConfiguration which suggests that some other object should have that responsibility.

It's the same issue as with `ISequentialCallContext.CheckNextCall`. We don't want to make it public, but it has to be... The alternative is to expose the `AfterMustHaveHappenedConfiguration ` class directly, but it's consistent with the rest of the API where we usually expose interfaces.

A possible solution is to expose the method in the interface, but hide it with `[EditorBrowsable(Never)]` and document it to be "conceptually internal", as [in #605](https://github.com/FakeItEasy/FakeItEasy/pull/605/files#diff-00e69038fb19194983b078aa099c04ceR13)